twpca/model.py

```diff
@@ -68,19 +76,18 @@ def __init__(self, data, n_components,
 
         # data dimensions
         self.n_trials, self.n_timepoints, self.n_neurons = self.data.shape
-        if n_components > self.n_neurons:
+        if self.n_components > self.n_neurons:
             raise ValueError('TWPCA does not support models with more components than neurons')
-        self.n_components = n_components
 
         # mask out missing data
-        self.mask = np.isfinite(self.data).astype(np.float32)
-        self.num_datapoints = np.sum(self.mask)
-        self._mask = tf.constant(self.mask)
+        mask = np.isfinite(self.data).astype(np.float32)
+        num_datapoints = np.sum(mask)
+        _mask = tf.constant(mask)
```

Comment:
We don't use any of these variables outside of `__init__`, so why store them as class attributes?